packages/connect-form/src/utils/validation.ts

```diff
@@ -345,17 +359,45 @@ function validateTLSAndHostWarnings(
 ): ConnectionFormWarning[] {
   const warnings: ConnectionFormWarning[] = [];
 
-  const hasNonLocalhost = !!connectionString.hosts
-    .map((host) => host.split(':')[0])
-    .find(
-      (hostname) => !['localhost', '127.0.0.1', '0.0.0.0'].includes(hostname)
-    );
+  const nonLocalhostsCount = connectionString.hosts
+    .map((host) => !isLocalhost(host))
+    .filter(Boolean).length;
 
-  if (hasNonLocalhost && !isSecure(connectionString)) {
+  if (nonLocalhostsCount && !isSecure(connectionString)) {
     warnings.push({
       message:
         'Connecting without tls is not recommended as it may create a security vulnerability.',
     });
   }
   return warnings;
 }
+
+function validateSocksWarnings(
+  connectionString: ConnectionString
+): ConnectionFormWarning[] {
+  const warnings: ConnectionFormWarning[] = [];
+
+  const searchParams = connectionString.typedSearchParams<MongoClientOptions>();
+  const proxyHost = searchParams.get('proxyHost');
+
+  if (!proxyHost || isLocalhost(proxyHost)) {
+    return warnings;
+  }
+
+  if (searchParams.get('proxyPassword')) {
+    warnings.push({
+      message: 'Using proxy with password.',
+    });
+  }
+
+  const mongoHostCount = connectionString.hosts.filter((x) =>
+    isAtlas(x)
+  ).length;
+  if (mongoHostCount) {
+    warnings.push({
+      message: 'Using proxy with MongoDB service host.',
+    });
+  }
```

Comment:
Should this be changed to something closer to this? (if not, I’m not quite sure I understand how Atlas comes into play here)
```suggestion
  const localMongoHostCount = connectionString.hosts.filter((x) =>
    isLocalhost(x)
  ).length;
  if (localMongoHostCount) {
    warnings.push({
      message: 'Using remote proxy with local MongoDB service host.',
    });
  }
```

Reply:
thanks @addaleax. i wanted to reach about these messages, but you were faster.

Reply:
useless comment ever .. but even shorter:

```
if (connectionString.hosts.find(isLocalhost)) {
    warnings.push({
      message: 'Using remote proxy with local MongoDB service host.',
    });
  }
```